modules/payment/src/Form/PaymentAddForm.php

```diff
@@ -138,7 +137,7 @@ protected function buildPaymentGatewayForm(array $form, FormStateInterface $form
       ],
     ];
 
-    if ($this->order->getCustomerId() && $selected_payment_gateway->getPlugin() instanceof SupportsStoredPaymentMethodsInterface) {
+    if ($selected_payment_gateway->getPlugin() instanceof SupportsStoredPaymentMethodsInterface) {
```

Comment:
Why this change? We are only able to store and attribute payment methods to users. If the order is anonymous (as was the use case for this PR for a client) then there is no sense in selecting a stored method since we won't find it.

Reply:
I see now there's a note on the d.o. issue for storing payment profiles for anonymous users, but I'm unclear how this is really to be done? This would require changes deeper in Commerce core, e.g. [here](https://github.com/drupalcommerce/commerce/blob/878b37c85d7f9706f63344644501b97a46c2102d/modules/payment/src/PaymentMethodStorage.php#L73). Perhaps expanding the scope of this PR is not the right place to tackle that larger task?

Reply:
Yes, PaymentMethodStorageInterface::loadForOrder was added and used to load the payment method for the order in addition to any that might exist for the user.  The change may make sense in a separate issue--I just want to have fully functioning admin UI for order payments, and the changes all relate and are really not that hard to grok, imho.